### DISCUSSION TITLE
Support Postgres permissions

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#2290] [Feature Ideas]
We'd like to better integrate Postgres permissions into the Mathesar API + UI permission system. This means that Mathesar's permissions would be built on a combination of Postgres permissions, rather than being separately implemented.

---
*If you'd like us to build this feature, please comment below with more details about how you plan to use this functionality.*

**Comment by steckdenis** [comment]
Hello,

(first, thanks for the wonderful tool that Mathesar is!)

My use-case for this feature is also linked to #2289 and #2288. I'm working in a decently-sized group of people (about 50 people), and we regularly use dozens of Excel sheets to administrate personnel, projects, deliverables, outcomes, lectures and other organizational stuff. Finding these Excel sheets, producing reports (so only selecting a few columns) and doing joints is very very difficult for now. We are actively looking for a tool that has the exact features of Mathesar, but its permission system is currently what prevents us from using it.

Basically, we need a centralized data storage but for which certain users can only view or edit certain tables, views or specific columns (for the specific columns: we are okay with splitting tables in "sensitive information" and "non-sensitive information" versions). Our use-case is that the secretary must be able to see and edit the personal details of every member of the organization. But the business developer must only be able to list all our deliverables jointed with who worked on them, with the "who" only having a first name, last name and job title. The PR person must know who is the member responsible for a specific project, but not where they (the member) live. Ideally, members should be able to update their personal information, or current funding source, but not see or edit that information for other members.

I can extend the use-case if you want, but I think that this message is already quite long :-) .

**Reply by kgodey** [reply]
Thank you for the kind words @steckdenis! It's very helpful to know what you'd like to use Mathesar for and what we're lacking, it helps us decide what features to prioritize building.

I would love to know more about your use case! Would you be open to doing a call to talk more about it?

**Comment by kgodey** [comment]
Closing because this is now implemented.
